### PR TITLE
Fix all RedundantConditionGivenDocblockType psalm errors.

### PR DESCRIPTION
**Pull request by Nenglish7**
I will comment when this is ready to be merged.

### CODE REVIEW

**Comment by paragonie-scott** [comment]
I feel like this is chasing down a rabbit hole of issues.

File: src/Cookie.php [hunk 1]
```diff
@@ -82,11 +82,8 @@ public function fetch(string $name)
             return null;
         }
         try {
-            /** @var string $stored */
+            /** @var string $stored */ 
             $stored = $_COOKIE[$name];
```
Comment (azjezz) [inline src/Cookie.php]: revert. `$_COOKIE[$name]` can be an array. in that case, casting to a string will result in a notice + unexpected result.

https://3v4l.org/u4X5h

File: src/Stream/MutableFile.php [hunk 2]
```diff
@@ -183,14 +183,7 @@ public function readBytes(int $num, bool $skipTests = false): string
...
             /** @var string $read */
-            $read = \fread($this->fp, $bufSize);
-            if (!\is_string($read)) {
-                // @codeCoverageIgnoreStart
-                throw new FileAccessDenied(
-                    'Could not read from the file'
-                );
-                // @codeCoverageIgnoreEnd
-            }
+            $read = (string) \fread($this->fp, $bufSize);
```
Comment (azjezz) [inline src/Stream/MutableFile.php]: why this change ? an exception should make things clear.

**Comment by paragonie-security** [comment]
We've made a ton of upstream changes that breaks this PR, and we fixed the issues elsewhere. Thanks for the effort you put in.
